src/app/admin/use-get-data.ts

```diff
@@ -0,0 +1,26 @@
+//import { req } from '@/shared/client'
+import { Product } from '@prisma/client'
+//import { ProductAttr } from '@/shared/product-types'
+import { useCallback, useEffect, useState } from 'react'
+
+export function useGetData() {
+  const [products, setProducts] = useState<Product[]>([])
+  const getProducts = useCallback(async () => {
+    try {
+      const response = await fetch('/api/products', { method: 'GET' })
+      if (!response.ok) {
+        throw new Error('Failed to fetch products')
+      }
+      const productsData = await response.json()
```

Comment:
**I** need to fix some type shit here. looks like productsData is `any` which is wrong but that's not your fault. I'll fix this after you merge.